pyxem/utils/segment_utils.py

```diff
@@ -0,0 +1,312 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+from numpy.ma import masked_where
+
+import matplotlib.pyplot as plt
+
+from scipy.ndimage import distance_transform_edt, label, binary_erosion
+from scipy.spatial import distance_matrix
+from scipy.signal import convolve2d
+
+from skimage.feature import peak_local_max
+from skimage.filters import sobel, threshold_li
+from skimage.morphology import watershed, disk
+
+from sklearn.cluster import DBSCAN
+
+import warnings
+
+
+def norm_cross_corr(image, template):
+    """Calculates the normalised cross-correlation between an image
+    and a template at zero displacement.
+
+    Parameters
+    ----------
+    image: np.array
+        Image
+    template: np.array
+        Reference image
+
+    Returns
+    -------
+    corr : float
+        Normalised cross-correlation between image and template.
+    """
+    f, t = image - np.average(image), template - np.average(template)
+    with warnings.catch_warnings():
+        warnings.filterwarnings(action='error', category=RuntimeWarning)
+        try:
+            corr = np.sum(f * t) / np.sqrt(np.sum(f**2) * np.sum(t**2))
+        except (RuntimeError, RuntimeWarning, ZeroDivisionError):
```

Comment:
@pc494 is all this try except and warning catching yours? if so could you provide a couple of comments please

Reply:
It's not (this is what git blame is for) , but I'll edit.